specification/azure-kusto/resource-manager/Microsoft.Kusto/preview/2020-01-01-preview/kusto.json

```diff
@@ -0,0 +1,388 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "KustoManagementClient",
+    "version": "2020-01-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/listLanguageExtensions": {
```

Comment:
Why are these all POST APIs instead of modelling languageExtensions as a child resource of clusters?  With this design users will not be able to setup language extensions via template deployments

Reply:
We declare those APIs as preview since their performance isn't satisfied enough and we have long development trip util it will. Furthermore, the action can't invoked short time after cluster creation (which would happen if it would be modelling as a child resource) since the provisioning time would be very long (~40 minutes compare to ~10 minutes without the action).
It is not a resource, just a boolean property (for each supported language) that specifies if the language is enabled or not - you cannot have more than one "language resource" under cluster for each language name. 
I understand it's not aligned with ARM template deployment concept, but that is exactly why it is preview API. Once it would be ready to be part of provisioning without increasing provisioning time so much, we will add as a child resource or cluster property in a stable API version.

Reply:
Why is the performance different between a PUT and POST API?

Reply:
PUT is for create cluster resource which takes ~10 minutes.
POST (addLanguageExtensions) is a proxy action on cluster resource that modifies its state. This modification can't run on creation or even right after it. If it will, the operation could take up to 30-40 minutes. When we will reduce the action time length, we will implement it as part of PUT API.

Reply:
sounds good.!

---
In reply to: [360715708](https://github.com/Azure/azure-rest-api-specs/pull/8006#discussion_r360715708) [](ancestors = 360715708)